### PR TITLE
GH-993 Adds Image widget

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes GH-993

Image *source* publish param fills &lt;img **src=""**/&gt;
Image *sizing* publish param optional values:
 * "actual" = Display the image with its actual width:height ratio
 * "fit" = Stretch the image to fit its container
 * "custom" = Rely on the 'customWidth' and 'customHeight' parameters

@GordonSmith @dtsnell4 @rpastrana Please review

http://rawgit.com/jbrundage/Visualization/GH-993-ImageWidget/demos/dermatology.html?src/other/Image

Signed-off-by: Jay Brundage <jaman.brundage@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
You need one more:
"stretch" - stretch to fit (what you have for fit now)
"scale" - scale to fit - but it maintains aspect ratio so it will scale to fit either width or height which ever one is appropriate...

**Comment by jbrundage** [comment]
@GordonSmith "fit" will currently behave like your description of "scale" if "lockAspectRatio" = true

**Comment by GordonSmith** [comment]
I missed that when I looked at the code - sample looks nice...  I will review properly once I am home (am in airport currently).

File: src/other/Image.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["../common/HTMLWidget"], factory);
+    } else {
+        root.other_Image = factory(root.d3, root.common_HTMLWidget);
```
Comment (mlzummo) [inline src/other/Image.js]: dont need root.d3 (also would cause a mismatch in the non amd version

**Comment by dtsnell4** [comment]
Looks good

File: src/other/Image.js (hunk 1 above)
Comment (jbrundage) [inline src/other/Image.js]: Good point, will remove

**Comment by GordonSmith** [comment]
Fit is not working OMM:  
image

**Comment by jbrundage** [comment]
@GordonSmith I can't reproduce that bug... what browser were you using?

**Comment by jbrundage** [comment]
@GordonSmith I see what's happened. You changed the AbsoluteSurface to 800% width and 600% height. It's just expanding to fit that width/height.

**Comment by GordonSmith** [comment]
@jbrundage "fit" does not maintain aspect ratio:  
image

**Comment by jbrundage** [comment]
@GordonSmith It will if you enable "lockAspectRatio"

image

**Comment by GordonSmith** [comment]
Doh (my version didn't have a big red arrow to prevent me from looking like an idiot)

**Comment by dtsnell4** [comment]
Perhaps lockAspectRatio should be true by default?  I would think it would be the most common scenario.

**Comment by jbrundage** [comment]
Agreed, changing that now...

**Pull request merged**